src/main/java/com/kelinci/mainapp/api/UserController.java

```diff
@@ -37,27 +31,38 @@ public String getListOfRegisteredUsers() {
 
     @DeleteMapping(value = "/registered/delete")
     public void deleteOurUser() {
-        registeredUser = null;
-        confirmedUser = null;
+        ourUser = null;
         listOfUsers.clear();
     }
 
     @PostMapping(value = "/registration")
     public void register(@RequestBody RegisterRequest request) {
 
-        Integer random = (int) (1000000 * Math.random());
-        String mailCode = Integer.toString(random);
-        System.out.println(mailCode);
-
-        final OurUser ourUser = new OurUser(request.getMail(), mailCode, false);
-
-        //wysylamy tutaj maila
-
-        registeredUser = ourUser;
-        listOfUsers.add(registeredUser);
-
+        final OurUser userFromRequest = new OurUser(request.getMail());
+        //in case the list is empty - add the userFromRequest with its randomly generated mailCode
+        if (listOfUsers.isEmpty()) {
+            String mailCode = generateMailCode();
+            userFromRequest.setMailCode(mailCode);
+            ourUser = userFromRequest;
+            listOfUsers.add(userFromRequest);
+            //in case the list is not empty, stream all email addresses from listOfUsers into a new List: listOfAllEmails
+            //the list maps and collects only email addresses from listOfUsers
+            //eventually if listOfAllEmails does not contain the email from the incoming user, generate mailCode and add user
+
+        } else {
+            List<String> listOfAllEmails = listOfUsers.stream().map(OurUser::getMail).collect(Collectors.toList());
```

Comment:
Wow - fajnie sobie zaczynasz radzić. Niemniej jednak można to zrobić lepiej. Można wszystko zrobić w streamie.
Struktura może wyglądać tak:
final OurUser newOurUser = listOfUser.stream()
.filter ( elementZList -> tutajMusiszWymyślićWarunekLogiczny(Zwracający true/false) któryChceszSprawdzić (npSprawdzićWartosciMaila?)   )
.findFirst()
.orElse(GdyWarunekZFilterSprawilZeNieMaTakiegoElementuWLiscieToTutajMozeszStworzycOurUsera);


Do przemyślenia (wygooglowania) ten stream. Tak poświęć na niego 15 min. Jak będziesz miał kłopoty to się wdzwonimy (bo to pierwszy taki stream także liczę, że jednak zadzwonisz xD xP )

Reply:
tutaj faktycznie będzie potrzebny call